### PR TITLE
Trim whitespaces from device data

### PR DESCRIPTION
**Pull request by Jellyfrog**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: LibreNMS/Modules/Os.php [hunk 1]
```diff
@@ -129,6 +129,7 @@ private function handleChanges(\LibreNMS\OS $os): void
 
         echo trans('device.attributes.location') . ': ' . optional($device->location)->display() . PHP_EOL;
         foreach (['hardware', 'version', 'features', 'serial'] as $attribute) {
+            $device->$attribute = trim($device->$attribute);
```
Comment (murrant) [inline LibreNMS/Modules/Os.php:132]: This changes all nulls to ''

Comment (murrant) [reply LibreNMS/Modules/Os.php:132]: I'm don't know if that is good or bad. Thoughts?

Comment (Jellyfrog) [reply LibreNMS/Modules/Os.php]: It does cause an deprecation, fixed it.

**Comment by Jellyfrog** [comment]
@murrant Another way would be to trim on model-level, https://laravel.com/docs/9.x/eloquent-mutators#inbound-casting
Thoughts?

**Review by murrant — APPROVED** [review]

**Pull request merged**